proxy/middleware.go

```diff
@@ -30,7 +30,18 @@ func (p *Proxy) AuthenticateSession(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		ctx, span := trace.StartSpan(r.Context(), "proxy.AuthenticateSession")
 		defer span.End()
-		if err := p.authenticate(false, w, r.WithContext(ctx)); err != nil {
+		if err := p.authenticate(w, r.WithContext(ctx)); err != nil {
+			s, _ := sessions.FromContext(r.Context())
```

Comment:
Error value thrown away - should it be checked?

Reply:
Though safe in this context, the flow is pretty janky. Updated to instantiated err and state at the same time.